# 6.2 Law of Sines

## Definition

Oblique $\triangle \equiv$ Contains no right angles. 

Solving a $\triangle \equiv$ Finding the measure of all sidelengths and angles. 

## Notation

We use same notation as right triangles. (Standard ABC for Angles, abc for opposite sidelengths to letter corresponding angles.)

## Pre-requisit Recollection

Assuming $\mathbb{R}^2$:

1. The following produce congruent trianges. 
    - ASA, AAS, SAS, SSS
    
$\implies$ That there is only 1 triangle for the given information. 
    
2. The following case is ambigous with three outcomes:
    - SSA
        1. There are no congruent triangles.
        2. There is one congruent triange. 
        3. There are two congruent trianges. 

## When to use the Laws.  

The law of sines is used for: 
    - AAS or ASA
    - SSA
    
The law of cosines is used for:
    - SAS
    - SSS


## Law of Sines

To find angles: $\frac{\sin A}{a} = \frac{\sin B}{b} = \frac{\sin C}{c}$

To find sidelengths: $\frac{a}{\sin A} = \frac{b}{\sin B} = \frac{c}{\sin C}$


## Q: Where does Law of Sines come from? 

1. Observe: Acute Triangle ABC | C is the top angle, and angles A, B are the base angles. 

    - If one drops an altitude bisection from angle C to the line AB, to form two right triangles, then: 

    $\sin A = \frac{h}{b} \implies h = b\sin A $

    $\sin B = \frac{h}{a} \implies h = a\sin B $
    
    $\implies b\sin A = a \sin B $

    $\implies \frac{\sin A}{b} = \frac{\sin B}{b} $ 
    
    
2. Observe: Obtuse Triangle ABC | C is the top angle, and angles A, B are the base angles. 

    - This time, the altitude drop from angle C does not bisect the prexisting triangle. It drops outside of the angle C, to form a new, smaller triangle exterior to the triangle ABC. The external triangle has a right angle R, and forms the triangle C2, R, A2. 
    
    $ \sin B = \frac{h}{a} \implies h = a \sin B $
    
    $ A_2 = (180\deg - A_1) $
    
    $ \sin A_2 = \frac{h}{b} \implies h = b \sin A_2 $
    
    $ \implies h = b \sin A $ 

# 6.2 HW

## 12. Solve the SSA triangle. 
Indicate whether the given measurements result in no​ triangle, one​ triangle, or two triangles. Solve each resulting triangle. Round each answer to the nearest tenth.

$ B = 45\deg \\ b = 35 \\ c = 40 $ 

In [63]:
B = 45 #deg
b = 35
c = 40

### Precursory Analysis 

Since this is SSA, it is congruently ambiguous. That means that there may be none, one, or two trianlges. 

$ h = c \sin B \\\implies h = 40 \sin 45\deg \approx 28$

$ \implies h < b < c $

$ \implies 2 \triangle s $

### Find angle C_1

$ \frac{\sin c}{c} = \frac{\sin B}{b} \implies \sin C = \frac{c\sin B}{b} \\ \implies C = arcsin(\frac{c\sin B}{b} ) $

In [64]:
import numpy as np 

# Find sin B.
Sin_B = np.sin(B * np.pi / 180)
print("Sin_B =", Sin_B)

# Find sin C.
Sin_C_1 = c * Sin_B / b
print("Sin_C_1 =", Sin_C)

# Find angle C.
C_1_rad = np.arcsin(Sin_C) # Note: output in in rad. 

# Convert rad to deg
C_1 = np.rad2deg(Sin_C) #deg
print("C_1 =", C)


Sin_B = 0.707106781187
Sin_C_1 = 0.808122035642
C_1 = 46.3019819738


### Find angle C_2

$ C_2 = 180 - C_1 $

In [65]:
C_2 = 180 - C_1
print("C_2 =", C_2)

C_2 = 133.698018026


### Find angle A_1

$ A_1 = 180 - B - C_1 $

In [66]:
A_1 = 180 - B - C_1
print("A_1 =", A_1)

A_1 = 88.6980180262


### Find angle A_2

$ A_2 = 180 - B - C_2$

In [67]:
A_2 = 180 - B - C_2
print("A_2 =", A_2)

A_2 = 1.30198197379


### Find the Sides

In [68]:
# We Have Solved the angles for both triangles as follows. 

# Triangle 1: Acute
print("Triangle 1: \n A_1 = {}, C_1 = {} \n".format(A_1, C_1))

# Triangle 2: Obtuse
print("Triangle 2: \n A_2 = {}, C_2 = {}".format(A_2, C_2))

Triangle 1: 
 A_1 = 88.69801802620597, C_1 = 46.30198197379403 

Triangle 2: 
 A_2 = 1.3019819737940281, C_2 = 133.69801802620597


#### To find the missing side: 

Triangle 1: Acute

$ \frac{a_1}{\sin A_1} = \frac{b}{\sin b} \\ \implies a_1 = \frac{b \sin A_1}{\sin B} $

Triangle 2: Obtuse

$ \frac{a_2}{\sin A_2} = \frac{b}{\sin b} \\ \implies a_2 = \frac{b \sin A_2}{\sin B} $

In [69]:
a_1 = b * np.sin(A_1 * np.pi / 180) #input must be in rad. 
print("a_1 =", a_1)

a_2 = b * np.sin(A_2 * np.pi /180)
print("a_2 =", a_2)

a_1 = 34.9909638374
a_2 = 0.795267081838


In [76]:
# Completed trianges have the following values: 

# T1: Acute
print("T1 Acute: A = {}, B = {}, C = {}, \n \
         a = {}, b = {}, c = {}".format(A_1, B, C_1, a_1, b, c))

print("T2 Obtuse: A = {}, B = {}, C = {}, \n \
          a = {}, b = {}, c = {}".format(A_2, B, C_2, a_2, b, c))

T1 Acute: A = 88.69801802620597, B = 45, C = 46.30198197379403, 
          a = 34.99096383737586, b = 35, c = 40
T2 Obtuse: A = 1.3019819737940281, B = 45, C = 133.69801802620597, 
           a = 0.7952670818380554, b = 35, c = 40


## This is apparently, not correct. 

So, we will consolodate the pipeline into a single cell calculation to verify accuracy. 

In [81]:
import numpy as np 

# Given Values
B = 45 #deg
b = 35
c = 40


# Find sin B.
Sin_B = np.sin(B * np.pi / 180)
print("Sin_B =", Sin_B)

# Find sin C.
Sin_C_1 = c * Sin_B / b
print("Sin_C_1 =", Sin_C)

# Find angle C.
C_1_rad = np.arcsin(Sin_C_1) # Note: output in in rad. 

# Convert rad to deg
C_1 = np.rad2deg(C_1_rad) #deg
print("C_1 =", C)



C_2 = 180 - C_1
print("C_2 =", C_2)



A_1 = 180 - B - C_1
print("A_1 =", A_1)

A_2 = 180 - B - C_2
print("A_2 =", A_2)


# We Have Solved the angles for both triangles as follows. 

# Triangle 1: Acute
print("Triangle 1: \n A_1 = {}, C_1 = {} \n".format(A_1, C_1))

# Triangle 2: Obtuse
print("Triangle 2: \n A_2 = {}, C_2 = {}".format(A_2, C_2))


# Find missing sides for respective T's. 
a_1 = b * np.sin(A_1 * np.pi / 180) / np.sin(B * np.pi /180)#input must be in rad. 
print("a_1 =", a_1)

a_2 = b * np.sin(A_2 * np.pi /180) / np.sin(B * np.pi / 180)
print("a_2 =", a_2)


# Print Complete Values. 

# T1: Acute
print("T1 Acute: A = {}, B = {}, C = {}, \n \
         a = {}, b = {}, c = {}".format(A_1, B, C_1, a_1, b, c))

# T2: Obtuse
print("T2 Obtuse: A = {}, B = {}, C = {}, \n \
          a = {}, b = {}, c = {}".format(A_2, B, C_2, a_2, b, c))

Sin_B = 0.707106781187
Sin_C_1 = 0.808122035642
C_1 = 46.3019819738
C_2 = 126.087147066
A_1 = 81.0871470657
A_2 = 8.91285293427
Triangle 1: 
 A_1 = 81.08714706573124, C_1 = 53.91285293426875 

Triangle 2: 
 A_2 = 8.912852934268756, C_2 = 126.08714706573124
a_1 = 48.8997993756
a_2 = 7.66874311937
T1 Acute: A = 81.08714706573124, B = 45, C = 53.91285293426875, 
          a = 48.899799375550195, b = 35, c = 40
T2 Obtuse: A = 8.912852934268756, B = 45, C = 126.08714706573124, 
           a = 7.668743119373605, b = 35, c = 40


### ERRORS

Sidelength a_1 = 34 != 48.9 
Sidelength a_2 = 5.4 != 7.7 

NOTE: Corrected. Forgot the rest of the equation for sidelengths. It helps to divide by the rest of the formula. Dickshit. 

In [84]:
# Now we have a calculator for this SSA problem. 
import numpy as np 

# Given Values
B = 34 #deg
b = 7.1
c = 2.1


# Find sin B.
Sin_B = np.sin(B * np.pi / 180)

# Find sin C.
Sin_C_1 = c * Sin_B / b

# Find angle C.
C_1_rad = np.arcsin(Sin_C_1) # Note: output in in rad. 

# Convert rad to deg
C_1 = np.rad2deg(C_1_rad) #deg

C_2 = 180 - C_1

A_1 = 180 - B - C_1

A_2 = 180 - B - C_2

# Find missing sides for respective T's. 
a_1 = b * np.sin(A_1 * np.pi / 180) / np.sin(B * np.pi /180)#input must be in rad. 

a_2 = b * np.sin(A_2 * np.pi /180) / np.sin(B * np.pi / 180)


# Print Complete Values. 

# T1: Acute
print("T1 Acute: A = {}, B = {}, C = {}, \n \
         a = {}, b = {}, c = {}".format(A_1, B, C_1, a_1, b, c))

# T2: Obtuse
print("T2 Obtuse: A = {}, B = {}, C = {}, \n \
          a = {}, b = {}, c = {}".format(A_2, B, C_2, a_2, b, c))


T1 Acute: A = 136.47981352934116, B = 34, C = 9.520186470658835, 
          a = 8.743193376264692, b = 7.1, c = 2.1
T2 Obtuse: A = -24.479813529341158, B = 34, C = 170.47981352934116, 
           a = -5.261235571533512, b = 7.1, c = 2.1
